worker/grpc_requests.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	pb "github.com/koraygocmen/gravitational/jobscheduler"
+	"google.golang.org/grpc"
+)
+
+// registerWorker dials the scheduler GRPC server and registers
+// the calling worker with the worker's GRPC server address.
+// Worker's GRPC server address is later used by the scheduler to dial
+// worker to start/stop/query jobs.
+func registerWorker() {
```

Comment:
Having `registerWorker` to call `os.Exit` makes it harder for us to reason about the execution flow of the program, for example other goroutines will not have ability co clean up the state and gracefully shut down.

Instead, we usually recommend to propagate error state via channels or context, and keep os.Exit like code in one place, e.g. main.

Reply:
same applies to all other code including `deregisterWorker`

Reply:
I changed my code to send all errors to an error channel, which will be processed by the main function.